bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/LcnModuleHandler.java

```diff
@@ -0,0 +1,411 @@
+import org.openhab.binding.lcn.internal.common.LcnAddrMod;
+import org.openhab.binding.lcn.internal.common.LcnChannelGroup;
+import org.openhab.binding.lcn.internal.common.LcnException;
+import org.openhab.binding.lcn.internal.connection.Connection;
+import org.openhab.binding.lcn.internal.connection.ModInfo;
+import org.openhab.binding.lcn.internal.converter.AbstractVariableValueConverter;
+import org.openhab.binding.lcn.internal.converter.AngleConverter;
+import org.openhab.binding.lcn.internal.converter.Co2Converter;
+import org.openhab.binding.lcn.internal.converter.CurrentConverter;
+import org.openhab.binding.lcn.internal.converter.EnergyConverter;
+import org.openhab.binding.lcn.internal.converter.IdentityConverter;
+import org.openhab.binding.lcn.internal.converter.LightConverter;
+import org.openhab.binding.lcn.internal.converter.PowerConverter;
+import org.openhab.binding.lcn.internal.converter.TemperatureConverter;
+import org.openhab.binding.lcn.internal.converter.VoltageConverter;
+import org.openhab.binding.lcn.internal.converter.WindspeedConverter;
+import org.openhab.binding.lcn.internal.subhandler.AbstractLcnModuleSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleMetaAckSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleMetaFirmwareSubHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LcnModuleHandler} is responsible for handling commands, which are
+ * sent to or received from one of the channels.
+ *
+ * @author Fabian Wolter - Initial contribution
+ */
+@NonNullByDefault
+public class LcnModuleHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(LcnModuleHandler.class);
+    private @Nullable LcnAddrMod moduleAddress;
+    private Map<LcnChannelGroup, @Nullable AbstractLcnModuleSubHandler> subHandlers;
+    private List<AbstractLcnModuleSubHandler> metadataSubHandlers;
+    private Map<ChannelUID, @Nullable AbstractVariableValueConverter> converters;
+
+    public LcnModuleHandler(Thing thing) {
+        super(thing);
+
+        subHandlers = Collections.synchronizedMap(new HashMap<>());
+        metadataSubHandlers = Collections.synchronizedList(new LinkedList<>());
+        converters = Collections.synchronizedMap(new HashMap<>());
+    }
+
+    @Override
+    public void initialize() {
+        LcnModuleConfiguration localConfig = getConfigAs(LcnModuleConfiguration.class);
+        LcnAddrMod localModuleAddress = moduleAddress = new LcnAddrMod(localConfig.segmentId, localConfig.moduleId);
+
+        try {
+            // create sub handlers
+            ModInfo info = getPckGatewayHandler().getModInfo(localModuleAddress);
+            for (LcnChannelGroup type : LcnChannelGroup.values()) {
+                AbstractLcnModuleSubHandler newHandler = type.getSubHandlerClass()
+                        .getDeclaredConstructor(LcnModuleHandler.class, ModInfo.class).newInstance(this, info);
```

Comment:
Why not have each LcnChannelGroup enum instance include a AbstractLcnModuleSubHandler factory lambda method for creating class instances? That way you wouldn't need to use reflection here.